agent/agent.go

```diff
@@ -273,3 +274,40 @@ func generateAgentID() string {
 	}
 	return agentId.String()
 }
+
+func getLogPath() (string, error) {
+	if logPath, set := os.LookupEnv("SCOPE_LOG_ROOT_PATH"); set {
+		return logPath, nil
+	}
+	currentUser, _ := user.Current()
+	homeDir := currentUser.HomeDir
+	logFolder := ""
+
+	if runtime.GOOS == "windows" {
+		logFolder = fmt.Sprintf("%s/AppData/Roaming/scope/logs", homeDir)
+	} else if runtime.GOOS == "darwin" {
+		logFolder = fmt.Sprintf("%s/Library/Logs/Scope", homeDir)
+	} else if runtime.GOOS == "linux" {
+		logFolder = "/var/log/scope"
+	}
+	if logFolder != "" {
+		isOk := true
```

Comment:
Why having this variable?

Reply:
becase I want to first try if the folder exist, and If not exist try to create the folder, if that throws an error (ex: permissions) I want to run the final block of the method to return a temp folder instead.

So in order to not repeat the same code in two branches I just introduce that Boolean variable.